redisearch/src/main/java/site/ycsb/db/RediSearchClient.java

```diff
@@ -0,0 +1,327 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ * <p>
+ * RediSearch client binding for YCSB.
+ * <p>
+ * All YCSB records are mapped to a Redis *hash field*.
+ * For scanning we use RediSearch's secondary index capabilities.
+ */
+
+package site.ycsb.db;
+
+import redis.clients.jedis.*;
+import redis.clients.jedis.commands.ProtocolCommand;
+import redis.clients.jedis.util.JedisClusterCRC16;
+import redis.clients.jedis.util.SafeEncoder;
+import site.ycsb.*;
+import site.ycsb.workloads.CoreWorkload;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * YCSB binding for <a href="https://github.com/RediSearch/RediSearch/">RediSearch</a>.
+ * <p>
+ * See {@code redisearch/README.md} for details.
+ */
+public class RediSearchClient extends DB {
+  public static final String HOST_PROPERTY = "redisearch.host";
+  public static final String PORT_PROPERTY = "redisearch.port";
+  public static final String PASSWORD_PROPERTY = "redisearch.password";
+  public static final String CLUSTER_PROPERTY = "redisearch.cluster";
+  public static final String TIMEOUT_PROPERTY = "redisearch.timeout";
+  public static final String INDEX_NAME_PROPERTY = "redisearch.indexname";
+  public static final String INDEX_NAME_PROPERTY_DEFAULT = "index";
+  private JedisCluster jedisCluster;
+  private JedisPool jedisPool;
+  private Boolean clusterEnabled;
+  private int fieldCount;
+  private String fieldPrefix;
+  private String indexName;
+
+  public void init() throws DBException {
+    Properties props = getProperties();
+    int port = Protocol.DEFAULT_PORT;
+    String host = Protocol.DEFAULT_HOST;
+    int timeout = Protocol.DEFAULT_TIMEOUT;
+
+    String redisTimeoutStr = props.getProperty(TIMEOUT_PROPERTY);
+    String password = props.getProperty(PASSWORD_PROPERTY);
+    clusterEnabled = Boolean.parseBoolean(props.getProperty(CLUSTER_PROPERTY));
+    String portString = props.getProperty(PORT_PROPERTY);
+    indexName = props.getProperty(INDEX_NAME_PROPERTY, INDEX_NAME_PROPERTY_DEFAULT);
+    if (portString != null) {
+      port = Integer.parseInt(portString);
+    }
+    if (props.getProperty(HOST_PROPERTY) != null) {
+      host = props.getProperty(HOST_PROPERTY);
+    }
+    if (redisTimeoutStr != null) {
+      timeout = Integer.parseInt(redisTimeoutStr);
+    }
+
+    Jedis setupPoolConn;
+    JedisPoolConfig poolConfig = new JedisPoolConfig();
+    if (clusterEnabled) {
+      Set<HostAndPort> jedisClusterNodes = new HashSet<>();
+      jedisClusterNodes.add(new HostAndPort(host, port));
+      jedisCluster = new JedisCluster(jedisClusterNodes, timeout, timeout, 5, poolConfig);
+      setupPoolConn = jedisCluster.getConnectionFromSlot(1);
```

Comment:
@filipecosta90 What's happening here?

Reply:
@sazzad16 we just need one arbitrary node on the cluster to send the FT.CREATE (using coordinator), So I'm just picking the one with 1st slot...

Reply:
Okay. Slots are 0-based though.